Library/Homebrew/formula.rb

```diff
@@ -2363,6 +2369,15 @@ def build
       stable.build
     end
 
+    # @private
+    def build_flags
+      mod_name = to_s.split("::")[0..-2].join("::")
+      return [] if mod_name.empty?
+
+      mod = const_get(mod_name)
+      mod.const_get(:BUILD_FLAGS)
+    end
```

Comment:
What's going on here and why wasn't it needed before? I find this pretty opaque.

Reply:
Previously, the flags were all global in `Homebrew.args`. Here we set a constant inside the formula namespace with the passed down flags. That's really the best we can do here. It is still not really as scoped as it should be, and it can't be, since the flags already need to be available when the `Formula` subclass is evaluated. Simply put, you cannot load the same formula with different build flags.

Reply:
Ok, with that in mind can each line be commented more heavily and use some more meaningful variable names? Pretty hard to understand what's going on here without it.